src/Type/Proxy.purs

```diff
@@ -48,11 +48,140 @@
 -- | ```
 module Type.Proxy where
 
+import Prelude
+
 -- | Value proxy for kind `Type` types.
 data Proxy a = Proxy
 
+derive instance eqProxy :: Eq (Proxy a)
+
+derive instance functorProxy :: Functor Proxy
+
+derive instance ordProxy :: Ord (Proxy a)
+
+instance applicativeProxy :: Applicative Proxy where
+  pure _ = Proxy
+
+instance applyProxy :: Apply Proxy where
+  apply _ _ = Proxy
+
+instance bindProxy :: Bind Proxy where
+  bind _ _ = Proxy
+
+instance booleanAlgebraProxy :: BooleanAlgebra (Proxy a)
+
+instance boundedProxy :: Bounded (Proxy a) where
+  bottom = Proxy
+  top = Proxy
+
+instance commutativeRingProxy :: CommutativeRing (Proxy a)
+
+instance discardProxy :: Discard (Proxy a) where
```

Comment:
I mean, of course, it _is_ a unit type, but somehow it seems to convey more information.

That said, we can't _use_ that information in a pattern match, since we can't bind type variables, so actually I'm fine with this.